firefox_ui_tests/remote/security/test_ssl_status_after_restart.py

```diff
@@ -0,0 +1,123 @@
+
+    def tearDown(self):
+        try:
+            self.identity_popup.close(force=True)
+            self.browser.tabbar.close_all_tabs([self.browser.tabbar.tabs[0]])
+        except NoSuchElementException:
+            # TODO: A NoSuchElementException may be thrown here when the test is skipped
+            # as under xvfb.
+            pass
+        finally:
+            FirefoxTestCase.tearDown(self)
+
+    @skip_if_e10s
+    @skip_under_xvfb
+    # Bug 995801: Test certificate status after browser restart
+    def test_ssl_status_after_restart(self):
+        for item in self.test_data:
+            with self.marionette.using_context('content'):
+                self.marionette.navigate(item['url'])
+            self.verify_certificate_status(item)
+            self.browser.tabbar.open_tab()
+
+        '''
+        self.restart()
+
+        i = 0
+        for item in self.test_data:
+            self.browser.tabbar.switch_to(i)
+            self.verify_certificate_status(item)
+            i = i + 1
+        '''
+
+    def verify_certificate_status(self, item):
+        url = item['url']
+
+        # Check the favicon
+        # TODO: find a better way to check, e.g., mozmill's isDisplayed
+        self.assertEqual(self.browser.navbar.locationbar.favicon.get_attribute('hidden'), 'false',
+                         'Lock icon is visible in identity box for ' + url)
+
+        self.identity_popup.box.click()
+        Wait(self.marionette).until(lambda _: self.identity_popup.is_open)
+
+        # Check the type shown on the idenity popup doorhanger
+        self.assertEqual(self.identity_popup.popup.get_attribute('className'),
+                         item['type'],
+                         'Extended certificate is verified for ' + url)
+
+        # Check the identity label
+        self.assertEqual(self.identity_popup.organization_label.get_attribute('value'),
+                         item['identity'],
+                         'Identity name is correct for ' + url)
+
+        # Get the information from the certificate
+        cert = self.browser.tabbar.selected_tab.certificate
+
+        # Open the Page Info window by clicking the More Information button
+        page_info = self.browser.open_page_info_window(
+            lambda _: self.identity_popup.more_info_button.click())
+
+        if item['identity'] != '':
+            owner = cert['organization']
+        else:
+            owner = page_info.get_property('securityNoOwner')
+
+        try:
+            # Verify that the current panel is the security panel
+            self.assertEqual(page_info.deck.selected_panel, page_info.deck.security)
+
+            # Verify the domain listed on the security panel
+            self.assertIn(self.security.get_domain_from_common_name(cert['commonName']),
+                          page_info.deck.security.domain.get_attribute('value'),
+                          'Expected domain found in certificate for ' + url)
```

Comment:
The current check handles a wildcard certificate, and it's running fine for the -ov item. It checks that the **domain** displayed on page info's security tab matches the **domain** on the certificate, in its common name field. This is the same check that's used in both the mozmill and marionette -ev test. (The mozmill test's separate -ov check doesn't test the domain at all.)

The mozill -dv tests' regex code checks that the **hostname** displayed on page info's security tab matches the certificate's common name field as a whole, which can include either a specific host or a wildcard for the domain.

I'm not sure why we'd want to do something different for the -dv test than for the others. Do we need or want to check for hostname rather than just for domain for each item?